sdk/core/azure-core/azure/core/rest/_helpers.py

```diff
@@ -103,9 +103,21 @@ def set_urlencoded_body(data, has_files):
         default_headers["Content-Type"] = "application/x-www-form-urlencoded"
     return default_headers, body
 
-
-def set_multipart_body(files):
-    formatted_files = {f: _format_data_helper(d) for f, d in files.items() if d is not None}
+FileContent = Union[IO[bytes], bytes, str]
+FileTypes = Union[
+    # file (or bytes)
+    FileContent,
+    # (filename, file (or bytes))
+    Tuple[Optional[str], FileContent],
+    # (filename, file (or bytes), content_type)
+    Tuple[Optional[str], FileContent, Optional[str]],
+    # (filename, file (or bytes), content_type, headers)
+    Tuple[Optional[str], FileContent, Optional[str], Mapping[str, str]],
+]
+
+def set_multipart_body(data: Dict[str, Any], files: Union[Mapping[str, FileTypes], Sequence[Tuple[str, FileTypes]]]):
```

Comment:
What is data? It's not used, and technically a breaking chance. I assume it's a typo?

Reply:
not a breaking change bc it's in private code, has to do with supporting data and files together, but agreed that's beyond the scope of this pr